relayer/src/utils.ts

```diff
@@ -0,0 +1,46 @@
+import { EventRecord } from "@polkadot/types/interfaces/system";
+import { ParaHeadAndId, ParachainsMap } from "./types";
+import Parachain from "./parachain";
+import Target from "./target";
+import logger from "./logger";
+
+// TODO: consider moving to a separate utils module
+// TODO: implement tests
+export const getParaHeadAndIdFromRecord = ({ event }: EventRecord): ParaHeadAndId => {
+    // use 'any' because this is not typed array - element can be number, string or Record<string, unknown>
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const { paraHead, paraId } = (event.toJSON().data as Array<any>)[0]
+        .descriptor;
```

Comment:
Can you explain what this is and why not using proper types?

Reply:
`event.data` is an array of different types - strings, numbers, and an object which has `descriptor` property

Reply:
But nothing is validated here. You need to throw an error and show what the actual contents is like or else it will explode in random places later.